src/Proyecto26.RestClient/Helpers/HttpBase.cs

```diff
@@ -15,7 +15,22 @@ public static IEnumerator CreateRequestAndRetry(RequestHelper options, Action<Re
             {
                 using (var request = CreateRequest(options))
                 {
-                    yield return request.SendWebRequestWithOptions(options);
+                    var sendRequest  = request.SendWebRequestWithOptions(options);
+                    if (options.ProgressCallback == null)
+                    {
+                        yield return sendRequest;
```

Comment:
Is this going to work without *IEnumerator*? 😲

Reply:
Yes! since the web request is an `AsyncOperation` it already runs in background. `yield return sendRequest` would just wait until `sendRequest.isDone` is set to true. 

`StartCoroutine(IEnumerator)` does basically the same thing as an `AsyncOperation`.

Reply:
awesome! let me check this next weekend, sorry for the delay :)